src/app/ui/universal-dapp-ui.js

```diff
@@ -131,172 +115,160 @@ UniversalDAppUI.prototype.renderInstanceFromABI = function (contractABI, address
+        <label class="pt-2 border-top d-flex justify-content-start flex-grow-1">
+          Low level interactions with contract
+        </label>
+        <i aria-hidden="true"
+        title="A contract can have at most one receive function, declared using receive() external payable { ... } and one fallback function, declared using fallback () external [payable].
+        These functions cannot have arguments, cannot return anything. Receive is executed on a call to the contract with empty calldata on plain Ether transfers (e.g. via .send() or .transfer()).
+        If no such function exists, but a payable fallback function exists, the fallback function will be called on a plain Ether transfer. If neither a receive Ether nor a payable fallback function is present, the contract cannot receive Ether through regular transactions and throws an exception. 
+        Fallback is executed on a call to the contract if none of the other functions match the given function signature."
+        class="fas fa-info text-info my-3 mx-1"></i>
+      </div>
+      <div class="d-flex flex-column">
+        <div class="d-flex justify-content-end m-2 align-items-center">
+          <label class="mr-2 m-0">Calldata</label>
+          ${calldataInput}
+          <button id="deployAndRunLLTxSendTransaction" class="btn btn-sm btn-secondary" title="Send data to contract." onclick=${() => sendData()}>Transact</button>
+        </div>
+      </div>
+      <div>
+        ${llIError}
+      </div>
+    </div>
+  `
+
+  function sendData () {
+    let error = false
+    function setLLIError (text) {
+      llIError.innerText = text
+      if (text !== '') error = true
+    }
+
+    setLLIError('')
+    const fallback = self.udapp.getFallbackInterface(contractABI)
+    const receive = self.udapp.getReceiveInterface(contractABI)
+    const args = {
+      funABI: fallback,
+      address: address,
+      contractName: contractName,
+      contractABI: contractABI
+    }
+    let calldata = calldataInput.value
+    const amount = document.querySelector('#value').value
+    if (amount !== '0') {
+      // check for numeric and receive/fallback
+      if (!helper.isNumeric(amount)) {
+        setLLIError('Value to send should be a number')
+      } else if (!receive && !(fallback && fallback.stateMutability === 'payable')) {
+        setLLIError("In order to receive Ether transfer the contract should have either 'receive' or payable 'fallback' function")
+      }
+    }
+    if (calldata) {
+      if (calldata.length > 3 && calldata.substr(0, 2) === '0x') {
+        if (!helper.isHexadecimal(calldata.substr(2, calldata.length))) {
+          setLLIError('the calldata should be a valid hexadecimal value.')
+        }
+      }
+      if (!fallback) {
+        setLLIError("'fallback' function is not defined")
+      }
+    }
+    if ((calldata || amount !== '0') && !error) self.runTransaction(false, args, null, calldata, null)
```

Comment:
why `amount !== '0'`

can't you just do `return setLLIError(...)`  when check fails and `self.runTransaction(false, args, null, calldata, null)` at the end?

Reply:
and send an 'empty' transaction?

Reply:
what do you mean empty?
even if there is no data and no value, that's still a valid tx. @LianaHus

Reply:
lets do this in the second pr

Reply:
please include this now (that's more readable)